### ISSUE TITLE
Bug: tag after weave bracket crashes compiler

### ISSUE THREAD

**Issue by joningold** [inkle/ink#212]
`* [Hi] #oops`

also 

`* [Hi #oops ]` and `* Hi[]     #oops `

**Comment by Stratege** [comment]
The reason for the crash is found at https://github.com/inkle/ink/blob/master/inklecate/InkParser/InkParser_Choices.cs#L91-L98

innerContent is null in these cases, which is guarded against with all other accesses but this one. I don't know enough about the output format to know wether or not just adding the appropriate guards against nulls to the tags would do the trick or if there is something deeper wrong with the code (where the change would fix that crash, but not make the output valid).

**Comment by joethephish** [comment]
Thanks for investigating @Stratege!

Tags are a new addition, so the null-ness is almost certainly okay, it just needs to guard appropriately. The important thing is to just find the last non-null content list, that corresponds to `* first [second] third`, and add the tag(s) there.

Would you be up for trying to fix @Stratege?

**Comment by Stratege** [comment]
okay, so I investigated a bit how the parsing works.

if we have a triple (before,optional-only,after) akin to "a [b] c" then that gets parsed as (a,b,c)
now if we add tags in various places:
"* a #d [b] c" -> (a,null,null) + tag "#d [b] c", tag insertion into first tried and worked
"* a [b #d] c" -> (a,b,null) + tag "#d ] c", tag insertion into third tried and crashed
"* a [b] c #d" -> (a,b,c) + tag "#d", tag insertion into third tried and worked

this is merely a sidenote as to how tagparsing works and implies strongly that the second example given above is wrong.
Options are now:
1. Either create a third one with just the tag in the case of having an empty text after a weave style inline bracket
2. Insert it into the first existing of (first,second,third), note that this will still crash in the case of "* [] #d"

with that reasoning I've appended a pull request implementing the first option. See #216
